src/lib/actions/pannable.ts

```diff
@@ -156,9 +165,20 @@ export function drag(
   }
 
   return (event) => {
-    coords.update(($coords) => ({
-      x: bounded($coords.x + (axis !== 'y' ? event.detail.dx : 0), 'x'),
-      y: bounded($coords.y + (axis !== 'x' ? event.detail.dy : 0), 'y'),
-    }));
+    coords.update(($coords) => {
+      //if the coords are just one number, we check how to interpret this
+      if(typeof $coords === "number"){
+        if(axis==="y"){
+          return bounded($coords + (axis !== 'x' ? event.detail.dy : 0), 'y')
+        }
+        if(axis==="x"){
+        return bounded($coords + (axis !== 'y' ? event.detail.dx : 0), 'x')}
```

Comment:
Same as above. 

I guess this comes down to taste but I often find if/else clauses hard to read. I'd probably do something like this:

```javascript
const delta = axis === 'x' ? event.detail.dx : event.detail.dy;
return bounded($coords + delta, axis);
```

(only works of course if you're sure at this point that `axis` is not `'xy'`)